coral/src/app/components/documentation/DocumentationEditor.tsx

```diff
@@ -0,0 +1,121 @@
+  const [text, setText] = useState(documentation || "");
+
+  function saveDocumentation() {
+    // trim to remove unnecessary whitespace at end/start
+    const newDocumentation = text.trim();
+
+    // existing `documentation` needs to be trimmed to compare, since
+    // there we don't know if the value saved before migrating to React
+    // contains spaces at beginning / end
+    if (documentation && newDocumentation === documentation.trim()) return;
+    save(newDocumentation);
+  }
+
+  return (
+    <Box.Flex flexDirection={"column"} rowGap={"3"}>
+      <Box.Flex
+        alignSelf={"start"}
+        component={"section"}
+        aria-label={"Switch between edit and preview mode"}
+      >
+        <SegmentedControlGroup
+          onChange={(value: ViewMode) => setViewMode(value)}
+          value={viewMode}
+        >
+          <SegmentedControl aria-pressed={viewMode === "edit"} value={"edit"}>
+            Edit
+          </SegmentedControl>
+          <SegmentedControl
+            aria-pressed={viewMode === "preview"}
+            value={"preview"}
+          >
+            Preview
+          </SegmentedControl>
+        </SegmentedControlGroup>
+      </Box.Flex>
+
+      {viewMode === "edit" && (
+        <div className={classes.markdownEditor}>
+          <Box
+            borderColor={"grey-20"}
+            borderWidth={"1px"}
+            borderRadius={"2px"}
+            marginBottom={"2"}
+            className={classes.markdownSyntaxHighlight}
+          >
+            <label
+              className={"visually-hidden"}
+              htmlFor={"markdown-editor-textarea"}
+            >
+              Markdown editor
+            </label>
+            <textarea
+              id={"markdown-editor-textarea"}
+              aria-describedby={"editor-markdown-description"}
+              autoFocus
+              value={text}
+              onChange={(event) => setText(event?.target?.value)}
+              className={classes.markdownTextarea}
+            />
+            <div aria-hidden={"true"}>
+              <SyntaxHighlighter
+                language={"markdown"}
+                PreTag="div"
+                style={a11yLight}
+                wrapLongLines={true}
+                customStyle={{ minHeight: "35vh" }}
+              >
+                {text}
+              </SyntaxHighlighter>
+            </div>
+          </Box>
+          <Typography.SmallTextBold id={"editor-markdown-description"}>
+            We are supporting markdown following the{" "}
+            <a href={"https://commonmark.org/help/"}>CommonMark</a> standard.
+          </Typography.SmallTextBold>
+        </div>
+      )}
+
+      {viewMode === "preview" && <DocumentationView markdownString={text} />}
+
+      <Box.Flex colGap={"l1"} justifyContent={"end"} alignItems={"center"}>
+        <Button.Secondary onClick={cancel} loading={isSaving}>
+          {isSaving ? "Saving documentation" : "Cancel"}
```

Comment:
Do we need to display the `"Saving documentation"` text also on the Cancel button? I feel like having it in the submit button is enough :o

Reply:
It's not displayed anyway, only the loading spinner is visible. This is in case someone with screenreader tries to cancel and otherwise would not get information why it's disabled.

Reply:
Ah! I completely missed that. Then, nitpick:  I think actually the cancel button should not be in loading state when submitting? Because it's not "cancelling" that is in progress, only "submitting".

Reply:
Good point. Should we then disable the button when loading?

Reply:
Yes, as I see it, when the submit action is async, then after clicking submit:
- submit button is in loading state
- cancel button is in disabled state